homeassistant/components/iotmeter/__init__.py

```diff
@@ -0,0 +1,55 @@
+"""IoTMeter integration for Home Assistant."""
+
+import logging
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .coordinator import IotMeterDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+_LOGGER.setLevel(logging.DEBUG)
+
+PLATFORMS = [Platform.SENSOR]  # Platform.NUMBER
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up IoTMeter from a config entry."""
+    _LOGGER.debug("Setting up IoTMeter integration")
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = {}  # Ensure that the domain is a dictionary
```

Comment:
Please use entry.runtime_data. Check airgradient for an example. You should extend the ConfigEntry type with the type for the runtime_data.